### ISSUE TITLE
Should Be Able To Retreat To SZ With Only Enemy Subs (>=V3)

### ISSUE THREAD

**Issue by ron-murhammer** [triplea-game/triplea#3428]
Here is the save game to reproduce with: 
[temp.zip](https://github.com/triplea-game/triplea/files/2011126/temp.zip)

1. I tested just doing the SZ147 attack through SZ129 (no other attacks) and it lets me retreat back to SZ129 which has the UK sub in it.

2. Then I tested also attacking the sub in SZ129 but doing the SZ147 battle first and it also let's me retreat to SZ129.

3. Then I tested the same but doing SZ129 battle before SZ147 and it doesn't let me retreat then... I'm guessing it then thinks SZ129 is hostile since there is a battle there and this is the bug.

TWW is close to the Global 40 rules so this issue probably exists in that and v3. Here is the rule excerpts indicating this should be legal:
"Move all attacking land and sea units in that combat that are on the battle strip to a single adjacent friendly space from which at least 1 of the attacking land or sea units moved. In the case of sea units, that space must have been friendly at the start of the turn."
"Sea zones are either friendly or hostile. Friendly sea zones contain no surface warships (this doesn’t include submarines and transports) belonging to a power with which you are at war. Hostile sea zones contain surface warships belonging to a power with which you are at war. (The presence of a surface warship belonging to an enemy power with which you are not yet at war doesn’t make a sea zone hostile.)"

**Comment by simon33-2** [comment]
Tested this with Global 1940 and it works properly. You can retreat to a SZ with a sub.

**Comment by DanVanAtta** [comment]
Seems we can close if this works properly in: TWW
I'm curious if this is still an issue or was fixed.

**Comment by Cernelius** [comment]
@DanVanAtta I assume @simon33-2 just misunderstood (not sure how, as @ron-murhammer is being very clear), and @ron-murhammer overlooked the comment or didn't care to reply? Just trying to mind read, see if I'm right.

"You can retreat to a SZ with a sub", indeed, starting from v3 (as also @ron-murhammer said per point 1).

The problem at this issue is that you cannot retreat to a sea zone you moved through if you already resolved a battle, in that same sea zone, on the same phase.

Reproduction steps (not needing any save games):

Select "World War II v3 1941".

Set "Low Luck" on.

Start "World War II v3 1941".

Skip until Russians Combat Move.

Turn Edit Mode on.

Add 1 German battleship in 2 Sea Zone.

Add 1 Germans submarine in 3 Sea Zone.

Add 3 Russians battleships in 4 Sea Zone.

Turn Edit Mode off.

Move 2 Russians battleships to 3 Sea Zone.

Move 1 Russians battleship to 2 Sea Zone.

End Combat Move.

Choose to attack the submarine in 3 Sea Zone.

Resolve the battle in 3 Sea Zone.

Make 1 round of the battle in 2 Sea Zone: you will see you cannot retreat to 3 Sea Zone, while you should be able to (by reading the rules, at least, if @panther2 can confirm). Instead, in this case, you are forced keeping making combat rounds until at least one of the battleships is removed.

**Comment by Cernelius** [comment]
I'll add that this is really a bad bug, since, if the bugged retreat point is your only one, you will be automatically pushed forward resolving more rounds on the battle. Then good luck arguing with your opponent you wanted to retreat, instead, especially if you rolled dice that resoulted clearly in his favour. Definitely one of the worst bugs, but such a situation should happen rarely.

**Comment by simon33-2** [comment]
Oh alright. I've tried your scenario and there is indeed still a problem. In fact, regardless of the sub being killed you can't retreat there. So not completely the same problem, although I guess it could only happen with subs or transports because otherwise that territory couldn't be a potential retreat territory.

**Comment by simon33-2** [comment]
Still a problem on 1.10.13537

Matters not if you attack the 3 Sea Zone sub actually.

**Comment by simon33-2** [comment]
I can now reproduce the problem on the Global map in 1.10.13537. Perhaps I tested it wrongly before.

**Comment by Cernelius** [comment]
>Matters not if you attack the 3 Sea Zone sub actually.

You are correct, if you are talking about not attacking by ignoring.

@ron-murhammer we actually have 4 instances, not 3, of which 2 are wrong, not 1:

1. If you only move through a submarine, you can retreat (no problem).

2. If you move through a submarine and attack it (with other units), without resolving that battle yet, you can retreat (no problem).

3. If you move through a submarine and attack it (with other units), resolving that battle first, you cannot retreat (problem).

4. If you move through a submarine and have other units in that sea zone, by the end of your Combat Move phase\*, but you choose not to attack it, you cannot retreat (problem).

\*This should happen only if those sea units were already in there or were moved in there to offload units into a land battle.

@ron-murhammer I suggest updating the first post with case 4.

**Comment by panther2** [comment]
@Cernelius 
>Make 1 round of the battle in 2 Sea Zone: you will see you cannot retreat to 3 Sea Zone, while you should be able to (by reading the rules, at least, if @panther2 can confirm). Instead, in this case, you are forced keeping making combat rounds until at least one of the battleships is removed.

Don't forget that during Conduct Combat Phase all combat in each space takes place at the same time. So in your example the battles in seazones 2 and 3 take place at the same time, too.

**Comment by Cernelius** [comment]
@panther2 Yeah, I was just talking in TripleA (as well as boardgame) terms. I didn't mean to imply that the battle you resolve first actually happens before the other one, because you chose it first.

Though I have to say that all battles happening at the same time is not as absolute as all movements, since the battles you offloaded units into cannot be happening at the same time, as they can even never happen if all units in them are offloaded and you retreat all those transports, in one or more sea battles you have to resolve first. So I'm really dubious if you would say that the battle you offloaded units into actually happens at the same time as the battle the ships are making to be able to actually offload.

**Comment by panther2** [comment]
@Cernelius Yes, that is why the amphibious assault is stated in the rulebooks as  somehow special move/combat.

But what I wanted to say with my previous statement is that you cannot retreat into an at that time contested seazone. When in your above scenario you choose to attack the submarine in seazone 3, you cannot "at the same time" retreat a ship from seazone 2 into seazone 3.

**Comment by Cernelius** [comment]
>But what I wanted to say with my previous statement is that you cannot retreat into an at that time contested seazone. When in your above scenario you choosed to attack the submarine in seazone 3, you cannot "at the same time" retreat a ship from seazone 2 into seazone 3.

Ok, so did you notice that what @ron-murhammer is saying is that he believes the bug is in the case number 3, while what you are saying is that the case number 3 is correctly behaving, while the bug is in the case number 2, instead (the engine letting you retreat to a sea zone where a battle is happening on the same phase).

These are the @ron-murhammer points:

> 1. I tested just doing the SZ147 attack through SZ129 (no other attacks) and it lets me retreat back to SZ129 which has the UK sub in it.
> 2. Then I tested also attacking the sub in SZ129 but doing the SZ147 battle first and it also let's me retreat to SZ129.
> 3. Then I tested the same but doing SZ129 battle before SZ147 and it doesn't let me retreat then... I'm guessing it then thinks SZ129 is hostile since there is a battle there and this is the bug.

@panther2 I'll take your word, but I feel it is really inconsistent to say you can move through a sea zone with only enemy submarines/transports in it, but you cannot retreat to that same sea zone, if you are also attacking them (while you still can move through them). It would be more consistent, in my opinion, if you would also be forbidden moving through ignorable units you are attacking, then.

@ron-murhammer If now I'm actually finally understanding all correctly, looks like your guesses on the rulebook were wrong, according to what @panther2 says. So, you should probably edit the first post, then (also don't forget adding my point 4, that is still bugged anyways).

When fixing this bug, I would also **add a property for allowing retreating to a sea zone with enemy units that you can all ignore, regardless of them being attacked or not on the same phase**, as this is what I believe makes sense the most (the property would be for custom games only).

**Comment by Cernelius** [comment]
@panther2 To be absolutely sure we are on the same page, then, if I move 1 battleship through a sea zone with only 1 transport to attack 1 enemy battleship in another sea zone and, on the same combat move phase, I also attack that transport, then I am forbidden retreating the battleship to the sea zone with the transport, correct (thus I cannot retreat at all if I'm just doing a battle between 1 battleship and 1 battleship, in this case)?

**Comment by Cernelius** [comment]
And changing the title from "**Should Be Able To Retreat To SZ With Only Enemy Subs (>=V3)**" to something like "**Should Not Be Able To Retreat To SZ With Only Enemy Subs If Attacking Them Too (>=V3)**".

**Comment by simon33-2** [comment]
@panther2 
> But what I wanted to say with my previous statement is that you cannot retreat into an at that time contested seazone. When in your above scenario you choosed to attack the submarine in seazone 3, you cannot "at the same time" retreat a ship from seazone 2 into seazone 3.

Is that correct? The Anniversary rulebook states that it must have been a friendly sea zone at the start of the turn. Nothing about there being a battle there or not.

Or is there something else changed?

**Comment by panther2** [comment]
@simon33-2 Here another rule applies:
"You cannot move additional units into an embattled space once an attack has begun."

**Comment by Cernelius** [comment]
Maybe this has made for boardgame reasons, as I can see in the moment you return those units from the battleboard to the gameboard they would get mixed up with the ones in the sea zone you are retreating to, and it would be substantially impossible to set them apart from those that are going to make the other battle, that you may do at whatever point thereafter.

However, I think we can all agree that, like, a defenceless transport should realistically really not bar retreating of warships, just because something is getting around automatically sinking it, so I suggest a property be made available for custom games, not having this restrictions, but always allowing retreating to sea zones with only enemy ignorable units (like transports and submarines). Of course, based on personal opinions of realism we could add a virtually infinite number of custom properties. And probably off topic, as this would be a feature request.

Anyways, either someone updates the first post of this topic or I guess this report is invalid.

**Comment by panther2** [comment]
@Cernelius 
>However, I think we can all agree that, like, a defenceless transport should realistically really not bar retreating of warships, just because something is getting around automatically sinking it, so I suggest a property be made available for custom games, not having this restrictions, but always allowing retreating to sea zones with only enemy ignorable units (like transports and submarines). 

It is not primarily to an ignorable unit you would retreat to but to a battle. You would allow retreating to seazones based on battle results of prior sea battles of the same Conduct Combat Phase and based on the coincidence of the enemy leftover being for example a submarine or maybe another warship. 

Not to deny whatever property, just to clarify.

**Comment by simon33-2** [comment]
> @simon33-2 Here another rule applies:
> "You cannot move additional units into an embattled space once an attack has begun."

I'd want a ruling on that point. Has one been issued? The intention of that rule is that you can't add additional units to an attack after you've started rolling dice, either for this attack or for another attack. And since all combat movement is simultaneous, doesn't it follow that all battles are also simultaneous? The only real point would be that retreating units shouldn't be able to participate in battles in spaces they have retreated to.

I really hate the idea that a retreat option is valid or not based on whether you have resolved another combat or not.

**Comment by panther2** [comment]
> I really hate the idea that a retreat option is valid or not based on whether you have resolved another combat or not

No. The retreat option is valid or not based on engaging the enemy submarine or not.

**Comment by Cernelius** [comment]
@ron-murhammer Do you read?

Either this issue is edited correctly, or someone should close this as invalid/wrong. Then the issue preferably summarized correctly in a new one (assuming @panther2 is right, unless you want to go requiring an official ruling from the authorities of the rulebook you are referring to (the original authors etc.)).

@simon33-2 If you want a (more) official ruling, I guess you'll have to go searching for it yourself. I doubt it will come in here on its own.

**Comment by panther2** [comment]
@simon33-2 @Cernelius 

It is sometimes difficult for me to participate in discussions as my comments are usually considered "somehow official" on the one one hand but on the other hands my comments - in rare cases - can only be additions to the discussion.

In this special issue my comments rather are "additions to the discussion", adding what the rulebook offers to the discussed subject, as I cannot recall any official clarification on whether retreating into a contested seazone that has been friendly at the start of the turn does "overwrite" the principle rule of "You cannot move additional units into an embattled space once an attack has begun."

So I will discuss this with Krieghund.

Sorry, I had not been @ mentioned again - and I somehow lost sight of this issue.

**Comment by Cernelius** [comment]
Sure. Well, it's a pyramid. Also Krieghund rulings can, at any moment, be made invalid by the original author himself, but we know that is not going to happen, really. This is one of the many cases in which you cannot be mathematically sure based on the rulebook only. It is substantially like a religion; until the original founder is alive, you can always ask him, or whoever he appoints. Eventually, also these games will stop having an universally recognized referring person, then everyone will be free to read the books as he pleases, nobody around to say who is right and who is wrong, but, for now, we are lucky we can eventually get to know the one truth (and the accumulated quantity of internet discussions, if not lost, might eventually allow to cover everything).

However, knowing how much you are into the latest rulesets, I'm quite sure Krieghund will confirm what you said, also because I can see, playing the real boardgame, you would have really no way of avoiding mixing retreating units with units that are waiting to be moved to the battleboard, in case.

**Comment by Cernelius** [comment]
>I can see, playing the real boardgame, you would have really no way of avoiding mixing retreating units with units that are waiting to be moved to the battleboard, in case.

This would not be an issue if the only ignorable units were only transports (or if submarines would have been defenceless). Also, please, be sure to clarify that this matter applies exactly the same way in case of transports as well as submarines (**also the title of this issue should be changed on this regard, to say "submarines/transports", instead of "subs"**).

**Comment by panther2** [comment]
@Cernelius 
>Also, please, be sure to clarify that this matter applies exactly the same way in case of transports as well as submarines

Yes, I will do. Also it will be interesting to cover the situation when the attacker (no destroyer) decides to engange "the submarine" but the defending submarine chooses to submerge. Then we have an embattled territory where the defender decides to escape combat by submerging during step 2 of the general combat sequence.

**Comment by panther2** [comment]
@Cernelius 

Yes, I am in the process of carfully wording the request, hopefully making sure to include all aspects and thoughts.

**Comment by DanVanAtta** [comment]
@ron-murhammer are you working on this, is it fixed or still an outstanding issue?

**Comment by ron-murhammer** [comment]
Outstanding. But this exists in the current release and isn't a show stopper.

**Comment by DanVanAtta** [comment]
Cool, is it on your radar to fix this? If not, there is a lot of detail here, this issue problem would benefit from being re-summarized to have just the outstanding task and repro cases and referencing this issue. The barrier to entry is quite high, I don't think anyone else will pick this up because of that.

**Comment by ron-murhammer** [comment]
Not really high on my list and I would have to review all the posts as well as I think there was additional findings beyond my first post here.

**Comment by DanVanAtta** [comment]
Precisely, I don't think this ticket is actionable. It's getting to be a year old, what will change to make this become actionable that has not already happened? Meanwhile we contribute to one more item to the bug queue and a personal assignment, which devalues both and makes it okay to both. I recommend we take the time to create an action item from this or close it as inactionable (the criteria for when this can be closed is not readily apparent).

**Comment by Cernelius** [comment]
@DanVanAtta This is not past the clarification phase. We could go with what any of us (make a poll?) think it's the correct interpretation of the rulebook, but, for any degree of certainty (as the original rules are not that clear), here we are substantially waiting for @panther2 to get back to the "authorities", possibly to have the full correct and clear rules. If @panther2 is not doing it, someone else would need to get back to the original author of the "cloned" games or its appointee. Unless we want to pick a guess and go with it, making it the official TripleA rule, of course, which would be probably better than nothing, and a decent chance to get it "right".

About "actionability", unless a developer also wants to take the responsibility to "decide" the rules, there is nothing yet to be actioned here, and I believe we are currently waiting for @panther2 to get back, here.

**Comment by panther2** [comment]
Sorry for my oversight... 
I have asked Krieghund now and am awaiting his answer.

**Comment by panther2** [comment]
So here is my question:
>Today I kindly ask you for a clarification for all rulesets starting from and including Anniversary Edition:
>
>In a sea combat is the attacker allowed to retreat into an embattled seazone with only enemy submarine(s) and/or transport(s) that are under attack there, too?

>Let’s assume that the retreat route has been correctly established (Attacking fleet moves from SZ 1 through SZ 2 to attack an enemy in SZ 3, leaving one or more warships in SZ 2 to attack enemy submarine(s) and/or transport(s) there).
>
>It is further obvious that the seazone to retreat into (SZ 2) has been friendly at the start of the turn (only enemy submarines and/or transports inside). So the retreat rule on page 19 (Anniversary rulebook) is clearly fulfilled.
>
>On the other hand we know from the rules that all combat in each space takes place at the same time. Also we have the quite generic rule that once an attack has begun, you cannot move additional units into an embattled space (for example page 13, Anniversary rulebook).
>
>Does the decision to attack the enemy submarine(s) and/or transport(s) in SZ 2 affect the retreat rule? As this attack causes an embattled space in that seazone? Can we move retreating units into an embattled space, then?

And here is Krieghund's answer:

>I would say that the sea units can retreat into SZ 2, but they may not participate in the battle there after doing so. This is in accordance with all relevant rules, as the retreating units are technically retreating into the zone rather than moving into it, and their inability to participate in the battle obeys the spirit of the rule prohibiting the movement of further units into the space once the battle begins. Since the SZ 2 will be friendly regardless of the outcome of the battle there, there is no conflict regarding the status of the sea zone vis-à-vis the retreat.
>
>This may seem a little awkward, but the alternative is to force the SZ 2 battle to precede the SZ 3 battle in order to avoid retreating into an “active” space, and the result would be the same either way.

**Comment by panther2** [comment]
So units should be able to retreat to a seazone that contains only enemy subs/transports regardless of whether those have been attacked or not.

**Comment by Cernelius** [comment]
Ok, cool. So, in practice, both me and @simon33-2 were correct, in our guesses.

I also very much like this ruling, as, as I said, it is my favourite solution.

@panther2 Looks like the culprit, where you guessed wrong (but we were all guessing, of course), is that you considered retreating as a movement, while it is now clarified that retreating is not moving, so the rule prohibiting the movement of further units into the space once the battle begins is irrelevant.

@ron-murhammer Please update the OP, accordingly, as now all is clear (and any developers can fix it).

**Comment by panther2** [comment]
I like it better, too, but the wording of the rulebook offered the alternative solution. I am happy that this is sorted out, now.

**Comment by Cernelius** [comment]
I believe what I said at my 3rd post, in this issue, covers all the problems, namely:

> 1. If you only move through a submarine, you can retreat (no problem).
> 
> 2. If you move through a submarine and attack it (with other units), without resolving that battle yet, you can retreat (no problem).
> 
> 3. If you move through a submarine and attack it (with other units), resolving that battle first, you cannot retreat (problem).
> 
> 4. If you move through a submarine and have other combat units in that sea zone, by the end of your Combat Move phase, but you choose not to attack it, you cannot retreat (problem).

So, my suggestion is copy-pasting this in the original post.

In practice, you are correct on the case you made, except missing the number 4 case. My version is just a more generalized rendering, for the first 3 points.

Also, the matter is rather referring to "ignorable" units (both submarines and transports, in v3), rather than specifically "submarines", so change all occurrences of "submarine" with "ignorable unit" or "submarine/transport".

**Comment by DanVanAtta** [comment]
Slightly off-topic note, "About "actionability" -> it's really great to see movement on this issue. I think we have the right idea that we need to be able to do something about our issues, lest why do we keep them hanging around? 

Waiting for a dev to pick up a task, or waiting for questions to be considered and clarified are all in-flight actions, active progress, which is perfect to see. :100: Keep it up guys.

**Comment by DanVanAtta** [comment]
What remains to be done in this issue, is anyone able to summarize?
Is there discussion to be had on an appropriate fix? Are we identifying or have identified further error scenarios? Could someone please summarize.

**Comment by panther2** [comment]
Sea units should be able to retreat into sea zones containing only enemy submarines and/or enemy transports regardless whether those enemy units currently are under attack or not (as the respective sea zone has been friendly at the beginning of the turn). In case the sea battle has not already been fought, the units retreating there must be separated and my not join the battle.

**Comment by DanVanAtta** [comment]
@panther2 I just set up a combat scenario on WWII V3 1941 that looked like this:
![Screenshot from 2020-05-20 17-54-14](https://user-images.githubusercontent.com/12397753/82511757-f308e200-9ac2-11ea-8c2c-a590c8c28ccb.png)

Notice that a bship has travelled through Sz6 containing only a sub and transport to engage a bship in Sz3. A cruiser is sent to Sz3 to engage the sub & transport.

Fighting the bship battle first in Sz3, I was able to retreat into Sz6 and the retreating bship did not join the Sz6 cruiser vs transport  & sub battle.

Is this accurately the scenario that you have described? If so, is this actually working properly, or is there something more to it?

**Comment by Cernelius** [comment]
@DanVanAtta That is the correct behaviour. However, if you fight in 6 Sea Zone first, then the battleship in 3 Sea Zone loses the ability to retreat, and this is one of the two wrong behaviours at hand (I already summarized).

As I said, I think the opening post at this issue is in need of being edited or this issue closed are resummarized fully in a new one. There is really little to know here, at the end. Everything could be fully summarized in a few lines.

In practice, the opening post is correct, but needs to be integrated with the information that:
- This is the same for whatever units you can move through (not only submarines).
- The issue exists even if you choose to ignore the units you are moving through.

**Comment by panther2** [comment]
@DanVanAtta As @Cernelius said.

**Comment by DanVanAtta** [comment]
Indeed, the battle ordering was an important detail and I think speaks to
having issues be very terse and generally lacking most discussion. This
would be a good issue to re-summarize and provide an updated save with
minimum repro steps
